test/functional/ex_cmds/mksession_spec.lua

```diff
@@ -37,22 +37,107 @@ describe(':mksession', function()
 
     -- Create three windows: first two from top show same terminal, third -
     -- another one (created earlier).
+    command('set sessionoptions+=terminal')
     command('terminal')
     command('split')
     command('terminal')
     command('split')
-    command('mksession '..session_file)
+    command('mksession ' .. session_file)
     command('%bwipeout!')
 
     -- Create a new test instance of Nvim.
     clear()
     -- Restore session.
-    command('source '..session_file)
+    command('source ' .. session_file)
 
     eq(funcs.winbufnr(1), funcs.winbufnr(2))
     neq(funcs.winbufnr(1), funcs.winbufnr(3))
   end)
 
+  it(
+    'do not restore :terminal if not set in sessionoptions, terminal in current window #13078',
+    function()
+      command('set sessionoptions-=terminal')
+
+      local tmpfile_base = file_prefix .. '-tmpfile'
+
+      command('edit ' .. tmpfile_base .. '1')
+      command('terminal')
+
+      local buf_count = #meths.list_bufs()
+      eq(2, buf_count)
+
+      eq('terminal', meths.buf_get_option(0, 'buftype'))
+
+      command('mksession ' .. session_file)
+      command('%bwipeout!')
+
+      -- Create a new test instance of Nvim.
+      clear()
+
+      -- Restore session.
+      command('source ' .. session_file)
+
+      -- no terminal should be set. As a side effect we end up with a blank buffer
+      eq(2, #meths.list_bufs())
+      eq('', meths.buf_get_option(meths.list_bufs()[1], 'buftype'))
+      eq('', meths.buf_get_option(meths.list_bufs()[2], 'buftype'))
+    end
+  )
+
+  it('do not restore :terminal if not set in sessionoptions, terminal hidden #13078', function()
+    command('set sessionoptions-=terminal')
```

Comment:
how much is different between these tests? any way to have a shared function to avoid lots of duplicate code?

Reply:
ah good point!

I guess I can do some cleanup. Do we have any code style that mentions nested `describe` in test files? I can also create a helper in `test/functional/helpers.lua` if that's better.

Reply:
nested describe should be avoided. Simple solution would be just to create a `local function test_foo()` that is called by the three new tests added here.

Reply:
cool thanks! I will do this later today, I will also fix the CI errors

Reply:
done in 5ef0ce342e05094c2b0c0db0edd7abbc78e57172 lmk if you have any suggestions for more cleanups. thanks